clpy/core/core.pyx

```diff
@@ -3170,8 +3169,8 @@ cpdef ndarray _diagonal(ndarray a, Py_ssize_t offset=0, Py_ssize_t axis1=0,
     cdef Py_ssize_t ndim = a.ndim
     if (axis1 >= ndim or abs(axis2) >= ndim or
             axis1 < - ndim or axis2 < - ndim):
-        raise ValueError('axis1(={0}) and axis2(={1}) must be within range '
-                         '(ndim={2})'.format(axis1, axis2, ndim))
+        raise numpy.AxisError('axis1(={0}) and axis2(={1}) must be within '
```

Comment:
Why is this needed? Original CuPy implementation is wrong?

Reply:
https://github.com/fixstars/clpy/blob/ccd1d34e1fc7ca762489609819aeb9e0c8147942/tests/clpy_tests/indexing_tests/test_indexing.py#L64-L72

These test needs error as same as numpy.
And numpy raise numpy.AxisError in diagonal.
So I change this error.

I will investigate this test in cupy.

Reply:
In numpy 1.15.4 , diagonal did not raise ValueError but numpy.core._internal.AxisError...
This is different from [numpy documentation](https://docs.scipy.org/doc/numpy-1.15.0/reference/generated/numpy.diagonal.html#numpy.diagonal).
How do I have to treat it?

Reply:
As [CuPy's document](https://docs-cupy.chainer.org/en/v2.5.0/install.html) said, the supported versions are 1.9 <= 1.13. So please try with v1.13.

#156 is not sufficient. (As I remember, the specified version is what that test is passed but not in documents...? @snsato forgot to clear the reason on the issue/PRs... please don't forget in the next time!)

Reply:
In numpy 1.13.0 , diagonal raise ValueError.
So I will revert Error change commit and change numpy version in CI